test/interpolating/query-params.test.js

```diff
@@ -0,0 +1,66 @@
+var test  = require('tape')
+  , rhumb = require('../../src/rhumb')
+
+test('Interpolating should handle zero parts', function (t) {
+  t.plan(2)
+
+  t.equal(rhumb.interpolate('?q=value', {}), '/?q=value'
+    , 'adds a slash when being interpolated')
+
+  t.equal(rhumb.interpolate('/?q=value', {}), '/?q=value'
+    , 'keeps the slash when being interpolated')
+})
+
+test('Interpolating should handle a fixed path with parts', function (t) {
+  t.plan(2)
+
+  t.equal(rhumb.interpolate('/foo?q=value', {}), '/foo?q=value'
+    , 'returns path without modification when being interpolated')
+
+  t.equal(rhumb.interpolate('/foo/bar/bing?q=value', {}), '/foo/bar/bing?q=value'
+    , 'returns path without modification when being interpolated')
+})
+
+test('Interpolating should handle a optional path with parts', function (t) {
+  t.plan(2)
+
+  t.equal(rhumb.interpolate('/wibble(/foo)?q=value', {}), '/wibble/foo?q=value'
+    , 'returns path without modification when being interpolated')
+
+  t.equal(rhumb.interpolate('/wibble(/foo/bar)?q=value', {}), '/wibble/foo/bar?q=value'
+    , 'returns path without modification when being interpolated')
+})
+
+test('Interpolating should handle a nested optional path with parts', function (t) {
+  t.plan(2)
+
+  t.equal(rhumb.interpolate('/wibble(/foo(/bing))?q=value', {}), '/wibble/foo/bing?q=value'
+    , 'returns path without modification when being interpolated')
+
+  t.equal(rhumb.interpolate('/wibble(/foo/bar(/bing))?q=value', {}), '/wibble/foo/bar/bing?q=value'
+    , 'returns path without modification when being interpolated')
+})
+
+test('Interpolating should handle multiple query params', function (t) {
+  t.plan(2)
+
+  t.equal(rhumb.interpolate('/wibble?foo=bar&bing=value', {}), '/wibble?foo=bar&bing=value'
+    , 'returns path without modification when being interpolated')
+
+  t.equal(rhumb.interpolate('/wibble/wobble?foo=bar&bing=value', {}), '/wibble/wobble?foo=bar&bing=value'
+    , 'returns path without modification when being interpolated')
+})
+
+test('Interpolating should handle incomplete query params', function (t) {
+  t.plan(1)
+
+  t.equal(rhumb.interpolate('/wibble?foo=bar&bing&', {}), '/wibble?foo=bar&bing&'
+    , 'returns path without modification when being interpolated')
+})
+
+test('Interpolating should handle empty query params', function (t) {
+  t.plan(1)
+
+  t.equal(rhumb.interpolate('/wibble?foo=&bar&bing=value', {}), '/wibble?foo=&bar&bing=value'
+    , 'returns path without modification when being interpolated')
+})
```

Comment:
1. I am not sure interpolate should resolve relative paths.  Should it not just interpolate vars, partials optionals, nested (as well as encode vars and partials)?
1. Agreed. let us not deal with this yet.